src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/task/respondtoclaimcallbackhandlertasks/SetApplicantResponseDeadline.java

```diff
@@ -0,0 +1,665 @@
+        if (respondent2HasSameLegalRep(caseData)) {
+            // if responses are marked as same, copy respondent 1 values into respondent 2
+            handleBothRespondentsSameLegalRepResponse(
+                callbackParams,
+                caseData,
+                updatedData,
+                responseDate,
+                applicant1Deadline
+            );
+            // only represents 2nd respondent - need to wait for respondent 1 before setting applicant response deadline
+        } else if (solicitorRepresentsOnlyOneOfRespondents(callbackParams, RESPONDENTSOLICITORTWO)) {
+            handleSingleRespondentResponse(callbackParams, updatedData, responseDate, caseData, applicant1Deadline);
+        } else {
+            handleDefaultResponse(callbackParams, updatedData, responseDate, caseData, applicant1Deadline);
+        }
+
+        updatedData.isRespondent1(null);
+        assembleResponseDocuments(caseData, updatedData);
+        frcDocumentsUtils.assembleDefendantsFRCDocuments(caseData);
+
+        if (toggleService.isUpdateContactDetailsEnabled()) {
+            addEventAndDateAddedToRespondentExperts(updatedData);
+            addEventAndDateAddedToRespondentWitnesses(updatedData);
+        }
+
+        retainSolicitorReferences(callbackParams.getRequest().getCaseDetailsBefore().getData(), updatedData, caseData);
+
+        UnavailabilityDatesUtils.rollUpUnavailabilityDatesForRespondent(
+            updatedData,
+            toggleService.isUpdateContactDetailsEnabled()
+        );
+
+        updateClaimsDetailsForClaimDetailsTab(updatedData, caseData);
+
+        updateDQPartyIdsIfHmcEnabled(updatedData);
+
+        caseFlagsInitialiser.initialiseCaseFlags(DEFENDANT_RESPONSE, updatedData);
+
+        // casefileview changes need to assign documents into specific folders, this is help determine
+        // which user is "creating" the document and therefore which folder to move the documents
+        // into, when directions order is generated in GenerateDirectionsQuestionnaireCallbackHandler
+        updateDocumentGenerationRespondent2(callbackParams, updatedData, caseData);
+
+        if (getMultiPartyScenario(caseData) == ONE_V_TWO_TWO_LEGAL_REP
+            && isAwaitingAnotherDefendantResponse(caseData)) {
+
+            return AboutToStartOrSubmitCallbackResponse.builder()
+                .data(updatedData.build().toMap(objectMapper))
+                .build();
+        }
+
+        // these documents are added to defendantUploads, if we do not remove/null the original,
+        // case file view will show duplicate documents
+        nullPlaceHolderDocuments(updatedData, caseData);
+        return AboutToStartOrSubmitCallbackResponse.builder()
+            .data(updatedData.build().toMap(objectMapper))
+            .state("AWAITING_APPLICANT_INTENTION")
+            .build();
+    }
+
+    private void handleDefaultResponse(CallbackParams callbackParams,
```

Comment:
handle, process, etc is a bit generic for a private  method that is doing something very specific (it should be), be more descriptive with method names and you won't need comments.